spec/helpers/number_format_helper_spec.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+require "spec_helper"
+
+RSpec.describe NumberFormatHelper do
+  describe "#number_with_limit" do
+    [
+      { input: 10, opts: {}, result: "10" },
+      { input: 10.0, opts: {}, result: "10" },
+      { input: 17203230.09932, opts: { length_limit: 8, digits: 8, precision: 3 }, result: "0.172e8" },
```

Comment:
:green_circle:  Ruby allows inserting `_` into numbers to make them more readable. This doesn't change the represented number in any way, but can make code with large numbers more readable. E.g. in this line

```suggestion
      { input: 17_203_230.09932, opts: { length_limit: 8, digits: 8, precision: 3 }, result: "0.172e8" },
```

I could easily see that the number is in the tens of millions.

Reply:
good hint